hack/make.sh

```diff
@@ -23,22 +23,9 @@ export SKOPEO_PKG='github.com/containers/skopeo'
 export SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 export MAKEDIR="$SCRIPTDIR/make"
 
-# We're a nice, sexy, little shell script, and people might try to run us;
-# but really, they shouldn't. We want to be in a container!
```

Comment:
This is still somewhat a concern; we don’t want someone inattentive to run `make test-integration-local` on the main development machine.

Reply:
This may be a problem we can simply ignore, developers/users must assume some amount of risk when running `make` and tinkering with development-project scripts.  It can be tricky (in scripts) to distinguish between "some random development environment" and a VM.  It's slightly easier in a container, but likely an unnecessary complexity.

Is this behavior you strongly want to preserve?   Assuming so, I could add a check similar to before (a canary env. var.) for containers, or alternately confirm `$CI` is set to `true` (guaranteed by Cirrus and most other CI systems).  Worth?

Reply:
I feel somewhat strongly about having some safeguard, primarily so that `make` or `make check` (something users are a bit likely to just type without reading much documentation) is not disruptive; alternatively we could make sure the Makefile and documentation is (over time maintained to be) consistent and safe, and it seems significantly easier to me to code a safeguard :)

I don’t really care very much at all _how_ that safeguard works exactly. (The whole `hack/make.sh` is inherited/stripped-down from a very different project, probably with way too much abstraction for how little it does; the in-container check is here because it was inherited in that file. Other than that check, maybe we don’t really need that file much.)

I don’t think it’s necessary to protect against directly running `hack/*`; such users should read the scripts first.

---

A manual `make check` (run on a non-CI Linux system) should be possible, so that integration/system tests can be developed locally, and should create a container; so a Cirrus-_only_ solution is not ideal. Checking for ($in_container or $CI) would work, I guess with a comment about where $CI comes from.

---

(Historical note: The old design that the disruptive checks require a caller to set SKOPEO_CONTAINER_TESTS, and that was typically set by `make check` for code running inside a container; Travis ran all tests inside a container. A CI could just run `make test-integration-local SKOPEO_CONTAINER_TESTS=1` on a test VM directly. The is a bit misleading, it didn’t anticipate dedicated VMs, it’s just an opt-in to disruptive behavior.)

---

There are now two mechanisms: `SKOPEO_CONTAINER_TESTS` and `container_magic`, AFAICS both roughly comparably safe; we probably need only one, I don’t care which one.

Reply:
Thanks for the explanation and for your opinion.  It sounds like we can use SKOPEO_CONTAINER_TESTS, default that to 0 (disabled) in the Makefile unless `CI` is also set.  Conveniently that also provides a handy place to document the potential dangers of `SKOPEO_CONTAINER_TESTS=1`.

As for `hack/make` and friends, my preference is to leave them mostly alone for now.  Even if inefficient/overly complex, they do function as good detail-encapsulation for both humans and automation.  So I think it's perfectly safe to leave and overhaul of them to a future PR, after all the "big-picture" stuff is functioning.